src/python/pants/backend/jvm/rules/targets.py

```diff
@@ -276,29 +263,20 @@ class NetrcCredentials(Target):
 # -----------------------------------------------------------------------------------------------
 
 
-class JarsField(PrimitiveField):
+class JarsField(SequenceField):
     """A list of `jar` objects to depend upon."""
 
     alias = "jars"
+    expected_element_type = JarDependency
+    expected_type_description = "an iterable of `jar` objects (e.g. a list)"
     value: Tuple[JarDependency, ...]
     required = True
 
-    # TODO: factor out a common Sequence field.
     @classmethod
     def compute_value(
         cls, raw_value: Optional[Iterable[JarDependency]], *, address: Address
     ) -> Tuple[JarDependency, ...]:
-        value = super().compute_value(raw_value, address=address)
-        try:
-            ensure_list(value, expected_type=JarDependency)
-        except ValueError:
-            raise InvalidFieldTypeException(
-                address,
-                cls.alias,
-                value,
-                expected_type="an iterable of `jar` objects (e.g. a list)",
-            )
-        return tuple(value)
+        return cast(Tuple[JarDependency, ...], super().compute_value(raw_value, address=address))
```

Comment:
The `cast` is necessary because the superclass returns `Optional[Tuple[JarDependency, ...]]`, which is normally necessary but not true here since we set `required = True`. 

It would be really tricky to teach MyPy to inspect that `required = True` and then to correctly infer the more precise type hint, so we accept the clunkiness of a `cast`.